encodings/dict/src/compute/min_max.rs

```diff
@@ -0,0 +1,10 @@
+use vortex_array::compute::{MinMaxFn, MinMaxResult, min_max, take};
+use vortex_error::VortexResult;
+
+use crate::{DictArray, DictEncoding};
+
+impl MinMaxFn<&DictArray> for DictEncoding {
+    fn min_max(&self, array: &DictArray) -> VortexResult<Option<MinMaxResult>> {
+        min_max(&take(array.values(), array.codes())?)
```

Comment:
I think that if codes are nonnullable/all_valid, we can use the `min_max` of the values directly.

Reply:
the problem is that we don't slice the values so min_max from values might actually not exist in the array

Reply:
unfortunate but reasonable

Reply:
Another thing here - take on a dict returns a dict by `take`-ing the codes and reusing the values, I'm not sure how this implementation doesn't blow up the stack.